contrib/perflog

```diff
@@ -0,0 +1,129 @@
+    r'\[(?P<date>.*)\] \[.*\].* \[pid \d+:tid \d+\] \[remote .*\] '
+    r'ipa: DEBUG: \[jsonserver_session\] (?P<principal>\S+): '
+    r'(?P<command>\S+)/1\(.*\): SUCCESS etime=(?P<etime>\d+)'
+)
+DATE_FORMAT = '%a %b %d %H:%M:%S.%f %Y'
+
+logger = logging.getLogger(__name__)
+
+
+class parselog(admintool.AdminTool):
+    command_name = "parselog"
+
+    usage = "%prog [options]"
+    description = "Parse the Apache error log for performance data"
+
+    def __init__(self, options, args):
+        super(parselog, self).__init__(options, args)
+        self.times = {}
+        self.since = None
+
+    @classmethod
+    def add_options(cls, parser):
+        super(parselog, cls).add_options(parser, debug_option=True)
+        parser.add_option(
+            "--command",
+            dest="command",
+            action="store",
+            default=None,
+            help="Command to analyze",
+        )
+        parser.add_option(
+            "--start-time",
+            dest="start_time",
+            action="store",
+            default=None,
+            help="time to begin analyzing logfile from",
+        )
+        parser.add_option(
+            "--file",
+            dest="file",
+            action="store",
+            default="/var/log/httpd/error_log",
+            help="Log file to pasre",
+        )
+
+    def validate_options(self):
+        super(parselog, self).validate_options(needs_root=True)
+
+        if self.options.start_time:
+            self.since = datetime.strptime(
+                self.options.start_time,
+                DATE_FORMAT
+            )
+
+    def run(self):
+        super(parselog, self).run()
+
+        if not is_ipa_configured():
+            logger.error("IPA client is not configured on this system.")
+            raise admintool.ScriptError()
+
+        with open(self.options.file, 'r') as f:
+            data = f.read()
+
+        matches = list(re.finditer(TIME_RE, data))
+
+        command = self.options.command
+
+        for match in matches:
+            if self.since:
+                logtime = datetime.strptime(match.group('date'), DATE_FORMAT)
+                if logtime < self.since:
+                    continue
+            if command is None or match.group('command') == command:
+                cmd = match.group('command')
+                if cmd not in self.times:
+                    self.times[cmd] = []
+                self.times[cmd].append(float(match.group('etime')))
+
+        if self.times:
+            for command in self.times:
+                # Average dropping the min and max
+                if len(self.times[command]) > 5:
+                    meantime = mean(sorted(self.times[command])[1:-1])
+                    num = len(self.times[command]) - 2
+                else:
+                    meantime = mean(self.times[command])
+                    num = len(self.times[command])
+                print(
+                    'Mean %s: %s of %d executions' % (
```

Comment:
It would be nice to display the units (ns) in the output, and the %s format for the float meantime may display irrelevant precision, for instance: `Mean user_add: 276355054.92528737 of 174 executions`.
Maybe 'Mean %s ns: %.0f of %d executions' would be enough?